File: src/main/java/me/lokka30/levelledmobs/result/RuleCheckResult.java
package me.lokka30.levelledmobs.result;

/**
 * Holds rule results in order to preserve the chance rule state
 *
 * @author stumper66
 * @since 3.1.2
 */
public class RuleCheckResult {

    public RuleCheckResult(final boolean useResult) {
        this.useResult = useResult;
    }

    public RuleCheckResult(final boolean useResult, final Boolean ruleMadeChance) {
        this.useResult = useResult;
        this.ruleMadeChance = ruleMadeChance;
    }

    public final boolean useResult;
    public Boolean ruleMadeChance;
}

File: levelledmobs-plugin/src/main/kotlin/io/github/arcaneplugins/levelledmobs/result/RuleCheckResult.kt
package io.github.arcaneplugins.levelledmobs.result

/**
 * Holds rule results in order to preserve the chance rule state
 *
 * @author stumper66
 * @since 3.1.2
 */
class RuleCheckResult {
    var useResult = false
        private set
    var ruleMadeChance: Boolean? = null
        private set

    constructor(useResult: Boolean){
        this.useResult = useResult
    }

    constructor(useResult: Boolean, ruleMadeChance: Boolean?){
        this.useResult = useResult
        this.ruleMadeChance = ruleMadeChance
    }
}